### PR TITLE
Remove yield tests and other minor test tweaks

### LINKED ISSUE #5076: Remove yield tests
Getting the following warning from pytest

WC1 /Users/sbird/Dev/bokeh/bokeh/bokeh/models/tests/test_glyphs.py yield tests are deprecated, and scheduled to be removed in pytest 4.0

### PR DESCRIPTION
**Pull request by birdsarah**
Fixes #5076 

Also, updates a number of tests to use pytest style (ping @brittainhard) -  we have agreed to slowly move tests over to pytest style so although some tests don't conform to it all new tests should (this should have been picked up in code review).

Adds a marker `api_test` to the api diff test as the test leaves cruft behind in working directory - ping @brittainhard.

### CODE REVIEW

File: bokeh/models/tests/test_annotations.py [hunk 1]
```diff
@@ -67,14 +69,14 @@ def test_ColorBar():
...
     assert color_bar.minor_tick_out == 0
-    yield check_text_properties, color_bar, "title_", "10pt", "bottom", "italic"
-    yield check_text_properties, color_bar, "major_label_", "8pt", "middle", "normal", "center"
-    yield check_line_properties, color_bar, "major_tick_", "#ffffff"
-    yield check_line_properties, color_bar, "minor_tick_", None
-    yield check_line_properties, color_bar, "bar_", None
-    yield check_line_properties, color_bar, "border_", None
-    yield check_fill_properties, color_bar, "background_", "#ffffff", 0.95
-    yield (check_properties_existence, color_bar, [
+    check_text_properties(color_bar, "title_", "10pt", "bottom", "italic")
```
Comment (mattpap) [inline bokeh/models/tests/test_annotations.py]: I presume the new approach preserves original location, line number, etc.?

**Comment by birdsarah** [comment]
@mattpap, removing radius_dimension from Circle. Here's the error in master:

```
=========================================================================================================== FAILURES ============================================================================================================
________________________________________________________________________________________________________ test_Circle[3] _________________________________________________________________________________________________________

model = <bokeh.models.markers.Circle object at 0x10c3ecbe0>
props = (['radius', 'radius_units', 'radius_dimension'], ['x', 'y', 'size', 'angle', 'angle_units'], ['fill_color', 'fill_alpha'], ['line_color', 'line_width', 'line_alpha', 'line_join', 'line_cap', 'line_dash', ...], ['visible'])
expected = set(['angle', 'angle_units', 'fill_alpha', 'fill_color', 'line_alpha', 'line_cap', ...]), found = set(['angle', 'angle_units', 'fill_alpha', 'fill_color', 'line_alpha', 'line_cap', ...])
missing = set(['radius_dimension'])

    def check_properties_existence(model, *props):
        expected = set(chain(PROPS, *props))
        found = set(model.properties())
        missing = expected.difference(found)
        extra = found.difference(expected)
>       assert len(missing) == 0, "Properties missing: {0}".format(", ".join(sorted(missing)))
E       AssertionError: Properties missing: radius_dimension

bokeh/models/tests/utils/property_utils.py:24: AssertionError
```

Here's the error with this PR:
```
=========================================================================================================== FAILURES ============================================================================================================
__________________________________________________________________________________________________________ test_Circle __________________________________________________________________________________________________________

    def test_Circle():
        marker = Circle()
        check_marker_properties(marker)
        assert marker.radius is None
        check_fill_properties(marker)
        check_line_properties(marker)
        check_properties_existence(marker, [
            "radius",
            "radius_units",
            "radius_dimension",
>       ], MARKER, FILL, LINE, GLYPH)

bokeh/models/tests/test_glyphs.py:466:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

model = <bokeh.models.markers.Circle object at 0x10e576d68>
props = (['radius', 'radius_units', 'radius_dimension'], ['x', 'y', 'size', 'angle', 'angle_units'], ['fill_color', 'fill_alpha'], ['line_color', 'line_width', 'line_alpha', 'line_join', 'line_cap', 'line_dash', ...], ['visible'])
expected = set(['angle', 'angle_units', 'fill_alpha', 'fill_color', 'line_alpha', 'line_cap', ...]), found = set(['angle', 'angle_units', 'fill_alpha', 'fill_color', 'line_alpha', 'line_cap', ...])
missing = set(['radius_dimension'])

    def check_properties_existence(model, *props):
        expected = set(chain(PROPS, *props))
        found = set(model.properties())
        missing = expected.difference(found)
        extra = found.difference(expected)
>       assert len(missing) == 0, "Properties missing: {0}".format(", ".join(sorted(missing)))
E       AssertionError: Properties missing: radius_dimension

bokeh/models/tests/utils/property_utils.py:24: AssertionError
```

They appear to contain exactly the same information, so I believe we're good.

**Comment by mattpap** [comment]
> They appear to contain exactly the same information, so I believe we're good.

OK, just checking.

**Comment by birdsarah** [comment]
I'm glad you did, I had checked they were still erroring (sp?), but I hadn't checked any change in reporting.

**Pull request merged**